include/deal.II/numerics/fe_field_convolution_function.h

```diff
@@ -0,0 +1,280 @@
+   * perform the convolution, and a radius, that is used to scale the
+   * convolution kernel. This function may be called also for points
+   * outside the domain, where it decays to zero according to what cut off
+   * function you used to construct this object.
+   *
+   * An example usage is the following:
+   * @code
+   *
+   * // Generate two triangulations
+   * Triangulation<dim> tria_1;
+   * Triangulation<dim> tria_2;
+   *
+   * // Read the triangulations from files, or build them up, or get them
+   * // from some place.
+   * ...
+   *
+   * // Associate a dof handler and a solution to the first triangulation
+   * DoFHandler<dim> dh1 (tria_1);
+   * Vector<double> solution_1;
+   *
+   * // Do the same with the second
+   * DoFHandler<dim> dh2 (tria_2);
+   * Vector<double> solution_2;
+   *
+   * // Setup the system, assemble matrices, solve problems and get the
+   * // nobel prize on the first domain...
+   * ...
+   *
+   * // Create a convolution kernel, and a convolution radius
+   * Functions::CutOffFunctionC1<dim> kernel;
+   * const double radius = 2*h;
+   *
+   * // Now build the convolution of the first function, projected on the
+   * // second domain
+   * FEFieldConvolutionFunction<dim> fe_function_1 (dh_1, solution_1,
+   *                                                kernel, radius);
+   * VectorTools::project (dh_2, constraints_2, quad,
+   *                       fe_function_1, solution_2);
+   *
+   * // Or interpolated...
+   * Vector<double> solution_3;
+   * VectorTools::interpolate (dh_2, fe_function_1, solution_3);
+   *
+   * @endcode
+   *
+   * FEFieldConvolutionFunction is designed to be an easy way to get the results
+   * of your computations across different, possibly non matching, grids. No
+   * knowledge of the location of the points is assumed in this class.
+   *
+   * <h3>Using FEFieldConvolutionFunction with
+   * parallel::distributed::Triangulation</h3>
+   *
+   * When using this class with a parallel distributed triangulation object
+   * and evaluating the solution at a particular point, not every processor
+   * will own the cell at which the solution is evaluated. Rather, it may be
+   * that the cell in which this point is found is in fact a ghost or
+   * artificial cell (see
+   * @ref GlossArtificialCell
+   * and
+   * @ref GlossGhostCell).
+   * The solution may be evaluated on non owned cells, but there the solution
+   * will vanish at a distance greater than `radius` from the locally owned
+   * domain. If you want to have a consistent solution on distributed
+   * Triangulation objects, you have to make sure you sum the results from all
+   * processors, i.e., using Utilities::MPI::sum(). In this way you are
+   * guaranteed that the actual convolution is performed with a non-zero field
+   * on only one processor, and the sum of the results across all processors
+   * will produce the same value you'd get with a serial Triangulation.
```

Comment:
I think what you're saying is that the function `f(x)` above is extended by zero outside the locally owned domain, as far as the convolution is concerned? Maybe say so.